### PR TITLE
feat(Signin): add action to validate OTP signin

### PR DESCRIPTION
**Pull request by julianalucena**
Considers the following code validations:
* Existence (considering the informed `account_id`)
* Expiration
* Usage (was it already used?)

```bash
curl --request POST -v \    
  --url localhost:3000/signin/validate_otp \
  --header 'Accept: application/json' \
  --header 'Authorization: xalalaAuth\n' \
  --header 'Content-Type: application/json' \
  --data '{
    "account_id": "anVsaWFuYS5sdWNlbmFAaW5jb2duaWEuY29t",
    "code": "IiempWMXIXoc3TUgJH8kV3E7Wlo="
}'


# HTTP/1.1 200 OK
{
  "signup_id":"4ba6efab-b688-49da-8235-bc26d4278bf9",
  "signup_timestamp":1657563343,
  "structured_address":{
    "locale":"en-US",
    "country_name":"Brasil",
    "country_code":"BR",
    "state":"SP",
    "city":"São Paulo",
    "street":"Av. Paulista",
    "number":"1578",
    "postal_code":"01310200"
  }
}% 
```

### CODE REVIEW

File: spec/forms/signin/otp_form_spec.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        )
+      end
+    end
+  end
+
+  describe '#submit' do
+    subject(:submit) { form.submit }
+
+    context 'when attributes are valid' do
+      it 'updates sigin code used timestamp' do
```
Comment (hmenezes) [inline spec/forms/signin/otp_form_spec.rb:59]: :nail_care: 
```suggestion
      it 'updates signin code used timestamp' do
```

File: spec/forms/signin/otp_form_spec.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+      it 'returns the logged in user' do
+        expect(submit).to eq(signin_code.user)
+      end
+    end
+
+    context 'when attributes are invalid' do
+      let(:attrs) { {} }
+
+      it 'does not update sigin code used timestamp' do
```
Comment (hmenezes) [inline spec/forms/signin/otp_form_spec.rb:71]: :nail_care: 
```suggestion
      it 'does not update signin code used timestamp' do
```

**Review by hmenezes — COMMENTED** [review]
:+1:

File: spec/forms/signin/otp_form_spec.rb (hunk 2 above)
Comment (julianalucena) [reply spec/forms/signin/otp_form_spec.rb]: https://github.com/inloco/incognia-demo-api-ruby/pull/18/commits/161b09e224f1afb104bfdcd7e84bbb7210b33e49

File: spec/forms/signin/otp_form_spec.rb (hunk 1 above)
Comment (julianalucena) [reply spec/forms/signin/otp_form_spec.rb]: https://github.com/inloco/incognia-demo-api-ruby/pull/18/commits/161b09e224f1afb104bfdcd7e84bbb7210b33e49

**Review by hmenezes — APPROVED** [review]
:+1:

**Pull request merged**